Clarify Safari time picker behavior for `datetime-local`

mkurz opened the pull request:
#### Summary

Safari 14.1 and newer displays a time picker for `datetime-local`.
`datetime-local` was not working correctly in one of the Safari tech previews but was fixed before the final 14.1 release.

#### Test results and supporting details

See the Safari 14.1 release note (https://developer.apple.com/documentation/safari-release-notes/safari-14_1-release-notes)
> Added support for HTML date, time, and datetime-local input types in macOS, enabling date and time controls in forms.

This is confirmed by the according issue (comment 27): https://bugs.webkit.org/show_bug.cgi?id=214946#c27

More confirmations:
- With screenshot: https://github.com/Fyrd/caniuse/pull/5646#issuecomment-715180664
- https://github.com/Fyrd/caniuse/pull/5669

A jsfiddle to test yourself: https://jsfiddle.net/8gw2yk5e/


#### Related issues

- https://bugs.webkit.org/show_bug.cgi?id=214946#c27 (see comment 27)

mkurz committed:
Clarify note about Safari time picker for `datetime-local`

caugner commented:
> crazy, the screenshot in [Fyrd/caniuse#5646 (comment)](https://github.com/Fyrd/caniuse/pull/5646#issuecomment-715233820) shows the time picker...

I don't see a time _picker_ in those screenshots, only time _fields_.

caugner commented:
Closing as stale for now, but we can always reopen.

mkurz commented:
Wait a moment, I think the data still needs to be fixed. Someone has Safari available to check if things work in latest version?

mkurz commented:
Thanks, @caugner, you were right that the screenshots show time fields, not a time picker.

I checked upstream and found that the missing time picker is an intentional macOS design decision, as documented in [WebKit bug 261744](https://webkit.org/b/261744).
The current WebKit implementation also configures the macOS picker with only `NSDatePickerElementFlagYearMonthDay`, confirming that it remains date-only ([source](https://github.com/WebKit/WebKit/blob/826329072b7dc2eae74aa481a382224373ceee9e/Source/WebKit/UIProcess/mac/WebDateTimePickerMac.mm#L232-L233)).

I’ve rebased the PR and changed it to clarify the existing note rather than remove it.

The PR is ready for review again. Could you please take another look?

Elchi3 commented on html/elements/input/datetime-local.json:
```diff
@@ -32,7 +32,7 @@
               "safari": {
                 "version_added": "14.1",
-                "notes": "Safari only displays a date picker and does not display a time picker."
+                "notes": "Safari displays a date picker but not a time picker; requiring manual time entry is an intentional design decision. See [bug 261744](https://webkit.org/b/261744)."
```
```suggestion
                "notes": "Safari displays a date picker but not a time picker; entering time manually is an intentional design decision. See [bug 261744](https://webkit.org/b/261744)."
```

Elchi3 reviewed (approved):
This sounds good to me, thank you @mkurz! 👍 
I just reworded it slightly.

mkurz committed:
better wording

Co-authored-by: Florian Scholz <fs@florianscholz.com>

Elchi3 committed:
Merge branch 'main' into safari_14.1_datetime-local
